build/xtask/src/dist.rs

```diff
@@ -1423,12 +1488,10 @@ fn allocate_k(
 fn allocate_one(
     region: &str,
     size: u32,
+    align: u32,
     avail: &mut Range<u32>,
 ) -> Result<Range<u32>> {
-    // This condition is ensured by allocate_all.
-    assert!(size.is_power_of_two());
-
-    let size_mask = size - 1;
+    let size_mask = align - 1;
```

Comment:
If you're going to compute the mask this way, you still need to assert that `align.is_power_of_two()`, because the result is nonsense for non-power-of-two inputs.